### PR TITLE
Added LAYOUT_alice

### PR DESCRIPTION
**Pull request by MudkipMao**
I saw a FR for adding LAYOUT_alice and already had one made

## Description

Updated the layout in layouts/default/

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* https://github.com/qmk/qmk_firmware/issues/10435

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by MudkipMao** [comment]
One thing I noticed, but haven't fixed yet is that no other LAYOUT_xxx uses Fn/MO(1) however, in Alice layout, i typically has split right shift. What should I change this to?

File: layouts/default/alice/default_alice/keymap.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+     *        │ Shift  │ Z │ X │ C │ V │ B │        │ B │ N │ M │ , │ . │ / │  Shift  │ Fn│
+     *        ├─────┬──┴──┬┴───┴┬──┴───┴┬──┴──┐     ├───┴───┴──┬┴───┴┬──┴───┴──────┬──┴──┬┘
+     *        │ Ctl │     │ Alt │       │ GUI │     │          │ Alt │             │ Ctl │
+     *        └─────┘     └─────┴───────┴─────┘     └──────────┴─────┘             └─────┘
+     */
+    [0] = LAYOUT_alice(
+        KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,             KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPCE,
+        KC_PGUP, KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,                      KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,
+        KC_PGDN, KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,                      KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,
+                 KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,                      KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, MO(1),
```
Comment (MudkipMao) [inline layouts/default/alice/default_alice/keymap.c:21]: This is the line with MO(1), what should I map this to? Or should I make it full right-shift

**Comment by fauxpark** [comment]
How many/which boards would be applicable for this layout? From a cursory look at everything that mentions `alice`, it seems like there is no one standard "Alice" layout, so I don't think most of them would be able to take advantage of this.

**Comment by stanrc85** [comment]
There are a few that I know of, maybe more: projectkeyboard rev1 and rev2, sneakbox, OSA, upcoming rukia. 

On all the boards I've seen they have optional split rshift and split backspace.

File: layouts/default/alice/info.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
+{
```
Comment (zvecr) [inline layouts/default/alice/info.json:1]: If you enable "fast input" mode on the configurator, the order of keys is not correct.

**Review by zvecr — CHANGES_REQUESTED** [review]

**Comment by MudkipMao** [comment]
@zvecr How do I enable/test fast input mode? I have my info.json loaded in QMK Configurator but can't seem to find that option

**Comment by MudkipMao** [comment]
> How many/which boards would be applicable for this layout? From a cursory look at everything that mentions `alice`, it seems like there is no one standard "Alice" layout, so I don't think most of them would be able to take advantage of this.

Adding to @stanrc85's comment, Homerow Co's Wonderland PCB uses it as well

**Comment by fauxpark** [comment]
> There are a few that I know of, maybe more: projectkeyboard rev1 and rev2, sneakbox, OSA, upcoming rukia.

I guess I was just looking at all the wrong boards then. Looks good.

File: layouts/default/alice/default_alice/keymap.c (hunk 1 above)
Comment (fauxpark) [reply layouts/default/alice/default_alice/keymap.c:21]: IMO just double up RSFT.

File: layouts/default/alice/default_alice/keymap.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+     *  ┌┴──┬┘ ┌┴─────┼───┼───┼───┼───┼───┤          └┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴──────┴┐
+     *  │PDn│  │ Caps │ A │ S │ D │ F │ G │           │ H │ J │ K │ L │ ; │ ' │   Enter  │
+     *  └───┘ ┌┴──────┴┬──┴┬──┴┬──┴┬──┴┬──┴┐        ┌─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───────┬──┴┐
+     *        │ Shift  │ Z │ X │ C │ V │ B │        │ B │ N │ M │ , │ . │ / │  Shift  │ Fn│
+     *        ├─────┬──┴──┬┴───┴┬──┴───┴┬──┴──┐     ├───┴───┴──┬┴───┴┬──┴───┴──────┬──┴──┬┘
+     *        │ Ctl │     │ Alt │       │ GUI │     │          │ Alt │             │ Ctl │
+     *        └─────┘     └─────┴───────┴─────┘     └──────────┴─────┘             └─────┘
+     */
+    [0] = LAYOUT_alice(
+        KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,             KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPCE,
```
Comment (fauxpark) [inline layouts/default/alice/default_alice/keymap.c:18]:
```suggestion
        KC_ESC,  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,             KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by noroadsleft** [comment]
Currently supported keyboards that should be able to support this layout, based on my preliminary research:

- cheshire/curiosity
- ergosaurus 
- fallacy
- kb_elmo/sesame
- maartenwut/wonderland
- mechlovin/adelais (all revisions)
- projectkb/alice (both revisions)
- sck/osa
- sneakbox/aliceclone
- tgr/alice (duh)
- tkc/osav2
- xelus/valor

... and maybe:

- handwired/colorlice by @marhalloweenvt
  It's not clear whether this supports a 2u Backspace.
- handwired/owlet60 by @worthlessowl
  This is actually a 65%-ish layout by default, but the PCB's right column can be snapped off, in which case `LAYOUT_owlet60_60_percent_full_bsp` looks to be compatible.
- nightly_boards/alter/rev1 by @NightlyBoards
  It's not clear whether this supports a 2u Backspace either.

Layout macro aliases would likely be required on all of the supported boards.

**Comment by noroadsleft** [comment]
@MudkipMao Fast Input can be enabled by hitting `Ctrl+Shift+F`. Make sure to press Ctrl before Shift.

File: layouts/default/alice/info.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                {"x":4.55, "y":0.1},
+                {"x":5.55, "y":0.1},
+                {"x":6.55, "y":0.1},
+                {"x":7.55, "y":0.1},
+                {"x":9.75, "y":0.1},
+                {"x":10.75, "y":0.1},
+                {"x":11.75, "y":0.1},
+                {"x":12.75, "y":0.1},
+                {"x":14.75, "y":0.1},
+                {"x":15.75, "y":0.1, "w":2},
```
Comment (noroadsleft) [inline layouts/default/alice/info.json:25]: Fix to the key object ordering.

```suggestion
                {"x":0.4, "y":0},
                {"x":1.55, "y":0.1},
                {"x":2.55, "y":0.1},
                {"x":3.55, "y":0},
                {"x":4.55, "y":0.1},
                {"x":5.55, "y":0.1},
                {"x":6.55, "y":0.1},
                {"x":7.55, "y":0.1},
                {"x":9.75, "y":0.1},
                {"x":10.75, "y":0.1},
                {"x":11.75, "y":0.1},
                {"x":12.75, "y":0.1},
                {"x":13.75, "y":0},
                {"x":14.75, "y":0.1},
                {"x":15.75, "y":0.1, "w":2},
```

File: layouts/default/alice/default_alice/keymap.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline layouts/default/alice/default_alice/keymap.c:1]: There should be a license header here.

Sample:

https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: layouts/default/alice/info.json (hunk 4 above)
Comment (MudkipMao) [reply layouts/default/alice/info.json:25]: Does this fix the fast input mode issue? I tried repro'ing the issue but couldn't see anything wrong with it

**Comment by MudkipMao** [comment]
@zvecr I'm unable to repro the "fast input" mode issue on QMK configurator. Any tips on how to fix the issue? 
I am on MacOS and didn't see anything go wrong when I tried (both Safari and Firefox).

**Comment by fauxpark** [comment]
@MudkipMao when I turn on fast input and press Q three times, it selects these three keys:
image
when it should be:
image

**Comment by MudkipMao** [comment]
@fauxpark Ah I see what you mean now. I didn't understand what fast input mode did. I'll look into fixing this

File: layouts/default/alice/info.json (hunk 4 above)
Comment (MudkipMao) [reply layouts/default/alice/info.json:25]: Thanks I see now!

**Comment by MudkipMao** [comment]
Looks like the ctrl+shift+f doesn't work on MacOS, but I found the option under settings

**Review by zvecr — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by zvecr** [comment]
Thanks!

**Comment by stanrc85** [comment]
I think this may need a little more work. I'm adding my community layout and I believe the boards keyboard.h and rules.mk files would need to be updated to support this right? 

On tkc/osav2 I have to use "LAYOUT_all" to compile, but on projectkb/alice/rev2 I have to use "LAYOUT_default" in my layout/community/alice/stanrc85-alice/keymap.c file.

**Pull request merged**